src/document/DocumentCommandHandlers.js

```diff
@@ -1598,6 +1602,55 @@ define(function (require, exports, module) {
         ProjectManager.showInTree(MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE));
     }
 
+    /** 
+     * Show error dialog
+     * @param {string} err Error message to be displayed to the user
+     * @param {callback} callback
+     */
+    function showErrorDialog(err, callback){
+        callback = callback || function () {};
+        Dialogs.showModalDialog(
+            DefaultDialogs.DIALOG_ID_ERROR,
+            Strings.ERROR_DIALOG_HEADER,
+            StringUtils.format(
+                Strings.UNEXPECTED_ERROR,
+                err),
+            [{
+                className : Dialogs.DIALOG_BTN_CLASS_PRIMARY,
+                id        : Dialogs.DIALOG_BTN_OK,
+                text      : Strings.OK
+            }]
+        );
+        console.error(err);
+        callback();
```

Comment:
You don't want to wait until the dialog is dismissed?  If so, you can get a promise from the dialog and wait on that to finish.